<!DOCTYPE html>
<html lang="en">

<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta name="description" content="">
  <meta name="generator" content="Hugo 0.72.0" />

  <title>Introduction &middot; trusTEr</title>

    

  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pure/1.0.0/pure-min.css">

  <!--[if lte IE 8]>
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pure/1.0.0/grids-responsive-old-ie-min.css">
  <![endif]-->
  <!--[if gt IE 8]><!-->
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pure/1.0.0/grids-responsive-min.css">
  <!--<![endif]-->

  <!--[if lte IE 8]>
  <link rel="stylesheet" href="https://raquelgarza.github.io/github_pages_truster_test/css/side-menu-old-ie.css">
  <![endif]-->
  <!--[if gt IE 8]><!-->
  <link rel="stylesheet" href="https://raquelgarza.github.io/github_pages_truster_test/css/side-menu.css">
  <!--<![endif]-->

  <link rel="stylesheet" href="https://raquelgarza.github.io/github_pages_truster_test/css/blackburn.css">

  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.2/css/all.min.css">

  
  <link rel="preconnect" href="https://fonts.gstatic.com">
  <link href="https://fonts.googleapis.com/css2?family=Raleway&display=swap" rel="stylesheet" type="text/css">

  
  <script async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

 
  

  
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/10.6.0/styles/androidstudio.min.css">
  <script async src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/10.6.0/highlight.min.js"></script>
  
  <script>hljs.initHighlightingOnLoad();</script>
  

  <link rel="shortcut icon" href="https://raquelgarza.github.io/github_pages_truster_test/img/favicon.ico" type="image/x-icon" />

  
  

</head>


<body>
<div id="layout">

  
<a href="#menu" id="menuLink" class="menu-link">
  
  <span></span>
</a>
<div id="menu">

  
  <a class="pure-menu-heading brand" href="https://raquelgarza.github.io/github_pages_truster_test/">trusTEr</a>


  <div class="pure-menu">
    <ul class="pure-menu-list">
      
      
        <li class="pure-menu-item">
          <a class="pure-menu-link" href="https://raquelgarza.github.io/github_pages_truster_test/"><i class='fa fa-home fa-fw'></i>Home</a>
      
        </li>
      
      
        <li class="pure-menu-item">
          <a class="pure-menu-link" href="https://raquelgarza.github.io/github_pages_truster_test/post/"><i class='fa fa-list fa-fw'></i>Posts</a>
      
        </li>
      
      
        <li class="pure-menu-item">
          <a class="pure-menu-link" href="https://raquelgarza.github.io/github_pages_truster_test/about/"><i class='fa fa-user fa-fw'></i>About</a>
      
        </li>
      
      
        <li class="pure-menu-item">
          <a class="pure-menu-link" href="https://raquelgarza.github.io/github_pages_truster_test/contact/"><i class='fa fa-phone fa-fw'></i>Contact</a>
      
        </li>
      
    </ul>
  </div>

  <div class="pure-menu social">
  <ul class="pure-menu-list">

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
    <li class="pure-menu-item">
      <a class="pure-menu-link" href="https://linkedin.com/in/bioinfo-raquelgarza" rel="me" target="_blank"><i class="fab fa-linkedin fa-fw"></i>LinkedIn</a>
    </li>
    

    

    

    

    

    

    
    <li class="pure-menu-item">
      <a class="pure-menu-link" href="https://github.com/raquelgarza" rel="me" target="_blank"><i class="fab fa-github-square fa-fw"></i>GitHub</a>
    </li>
    

    

    

    

    

    

    

    

    

    

    

    

    

  </ul>
</div>


  <div>
  <div class="small-print">
    <small>&copy; 2023. All rights reserved.</small>
  </div>
  <div class="small-print">
    <small>Built with&nbsp;<a href="https://gohugo.io/" target="_blank">Hugo</a></small>
    <small>Theme&nbsp;<a href="https://github.com/yoshiharuyamashita/blackburn" target="_blank">Blackburn</a></small>
  </div>
</div>

</div>


  <div id="main">


<div class="header">
  <h1>Introduction</h1>
  <h2></h2>
</div>
<div class="content">

  <div class="post-meta">

  <div>
    <i class="fa fa-calendar fa-fw"></i>
    <time>121 Jan 313112, 00:00</time>
  </div>

  

  

  

</div>

  <h1 id="truster---a-trusting-te-cluster-analysis">trusTEr - A trusting TE cluster analysis</h1>
<p><strong>Version 0.1.1, written in Python 3.6.6</strong></p>
<p>Takes fastq files from 10x single cell RNA sequencing, clusters cells using Seurat, and can be used to produce
read count matrices in a cluster level. You can also quantify reads per cluster having predefined clusters.</p>
<h2 id="requirements">Requirements</h2>
<p>TrusTEr depends on several external software. We provide a Docker container and a conda environment for a quick-start.</p>
<p>TrusTEr requires:</p>
<ul>
<li>Cellranger</li>
<li>R (version 3.6)
<ul>
<li>Seurat</li>
</ul>
</li>
<li>TEtranscripts</li>
<li>STAR aligner</li>
<li>subset-bam and bamtofastq from 10x Genomics</li>
<li>Velocyto</li>
</ul>
<p>The package has been tested in Unix systems only and supports only SLURM job submissions.</p>
<h2 id="how-to-install">How to install</h2>
<h4 id="just-the-modules">Just the modules</h4>
<p>If you fulfill the requirements, you can install via pip:
<code>pip install truster</code></p>
<h4 id="with-docker-container">With Docker container</h4>
<h4 id="with-conda-environment">With conda environment</h4>
<h2 id="introduction">Introduction</h2>
<p>As single cell technologies haven&rsquo;t developed to the point where we can get the needed sequencing depth to study transposable elements expression, trusTEr seeks produce more reliable results by combining reads from closely related cells to gain information on a cell type level.</p>
<p><img src="https://raw.githubusercontent.com/ra7555ga-s/truster/main/img/approach.png" alt="approach"></p>
<h2 id="structure">Structure</h2>
<p>trusTEr uses composition assiciation to relate three main classes:</p>
<ul>
<li>Experiment: Includes information about the experiment as a whole. This is the main object you will be working with.
<ul>
<li>Name is required, description is optional.</li>
<li>Register samples by providing a path to its fastq files.</li>
</ul>
</li>
<li>Sample: Created by giving a path to fastq files
<ul>
<li>Name and ID required.</li>
</ul>
</li>
<li>Cluster: Created by running <code>get_clusters()</code> or <code>merge_samples()</code> functions (Or <code>set_clusters_outdir()</code> or <code>set_merge_samples_outdir()</code>).</li>
</ul>
<p><img src="https://raw.githubusercontent.com/ra7555ga-s/truster/main/img/compositionAssociation.png" alt="compositionAssociation"></p>
<h2 id="functionality-and-workflow">Functionality and workflow</h2>
<p>This package is meant to be run with the following workflow:</p>
<p><img src="https://raw.githubusercontent.com/ra7555ga-s/truster/main/img/workflow.png" alt="workflow"></p>
<p>Depending on the object type, you have access to different functions to go through these steps.</p>
<p><code>Experiment</code> is the main object one would work with. Here you will <code>register_sample()</code> or <code>register_samples_from_path()</code>.</p>
<p>An object of type <code>Sample</code> has access to step 7 and some handy wrappers to use <code>cellRanger</code>, perform clustering with Seurat and run and plot RNA velocity. These functions can be called for all registered samples from your object of type <code>Experiment</code> (See <code>quantify()</code>, <code>get_clusters_all_samples()</code>, <code>velocity_all_samples()</code>, <code>plot_velocity_all_samples()</code>).</p>
<p>The need for the class <code>Experiment</code> is clearer once the user wants to merge samples (See <code>merge_samples()</code>) or to run the same workflow for all the samples&rsquo; clusters.</p>
<p>The user won&rsquo;t work directly with an object of type <code>Cluster</code>, but this class includes all the functions needed to go through steps 1-6 of the workflow. Instead of running this pipeline individually for each cluster, one can run the workflow for each cluster of each registered sample using the <code>Experiment</code> function <code>process_clusters(mode = &quot;per_sample&quot;, ...)</code>.</p>
<p>One can also partition the workflow and run step by step in all registered samples or in a combined clustering using the transitioning functions of <code>Experiment</code> to call the needed functions in the class <code>Cluster</code> (See <code>tsv_to_bam_clusters()</code>, <code>filter_UMIs_clusters()</code>, <code>bam_to_fastq_clusters()</code>, <code>concatenate_lanes_clusters()</code>, <code>merge_clusters()</code>, <code>map_clusters()</code>, <code>TE_counts_clusters()</code>, <code>normalize_TE_counts()</code>).</p>
<h2 id="running-on-a-server">Running on a server</h2>
<p>At the moment of construction of an <code>Experiment</code> object, you can declare <code>slurm_path</code> and a <code>modules_path</code>. These files will be checked whenever a function can be run as an <code>sbatch</code> job.</p>
<p>For the moment, trusTEr only works with <code>slurm</code> systems. You need to create two json files:</p>
<p><strong>1. slurmPath</strong>
Declaring the <!-- raw HTML omitted -->sbatch options<!-- raw HTML omitted --> per function. You need to declare a field per function listed that you will use:</p>
<ul>
<li>quantify</li>
<li>get_clusters</li>
<li>merge_samples</li>
<li>tsv_to_bam</li>
<li>filter_UMIs</li>
<li>bam_to_fastq</li>
<li>map_cluster</li>
<li>TE_count</li>
<li>velocity</li>
<li>plot_velocity</li>
</ul>
<p>As a brief example:</p>
<pre><code>{
  &quot;__default__&quot; :
  {
      &quot;account&quot; : &quot;myaccount&quot;,
      &quot;time&quot; : &quot;00:15:00&quot;,
      &quot;nodes&quot; : 1,
      &quot;partition&quot; : &quot;dell&quot;,
      &quot;output&quot; : &quot;%j.out&quot;,
      &quot;error&quot; : &quot;%j.err&quot;,
      &quot;job-name&quot; : &quot;%j&quot;
  },
  &quot;quantify&quot; :
  {
      &quot;account&quot; : &quot;myaccount&quot;,
      &quot;time&quot; : &quot;10:00:00&quot;,
      &quot;nodes&quot; : 1,
      &quot;tasks-per-node&quot; : 20,
      &quot;partition&quot; : &quot;dell&quot;,
      &quot;output&quot; : &quot;%j.quantify.out&quot;,
      &quot;error&quot; : &quot;%j.quantify.err&quot;,
      &quot;job-name&quot; : &quot;%j.quantify&quot;
  },
  &quot;get_clusters&quot; : {...},
  &quot;merge_samples&quot; : {...},
  &quot;tsv_to_bam&quot; : {...},  
  &quot;filter_UMIs&quot; : {...},
  &quot;bam_to_fastq&quot; : {...},
  &quot;map_cluster&quot; : {...},
  &quot;TE_count&quot; : {...},
  &quot;velocity&quot; : {...},
  &quot;plot_velocity&quot; : {...}
}
</code></pre><p><strong>2. modulesPath</strong></p>
<p>In a <code>slurm</code> system, many times you need to load modules for a software to be available. In this json file you declare the name of the needed modules. You need to declare a field per function listed that you will use:</p>
<ul>
<li>quantify</li>
<li>get_clusters</li>
<li>merge_samples</li>
<li>tsv_to_bam</li>
<li>filter_UMIs</li>
<li>bam_to_fastq</li>
<li>map_cluster</li>
<li>TE_count</li>
<li>TE_count_unique</li>
<li>velocity</li>
<li>normalize_TE_counts</li>
<li>plot_TE_expression</li>
</ul>
<p>As an example:</p>
<pre><code>{
        &quot;quantify&quot;:[&quot;cellranger/3.1.0&quot;],
        &quot;velocity&quot;:[&quot;GCC/7.3.0-2.30&quot;, &quot;SAMtools/1.9&quot;, &quot;velocyto/0.17.17&quot;],
        &quot;get_clusters&quot;:[&quot;GCC/9.3.0&quot;, &quot;OpenMPI/4.0.3&quot;, &quot;Seurat/3.1.5-R-4.0.0&quot;],
        &quot;merge_samples&quot;:[&quot;GCC/9.3.0&quot;, &quot;OpenMPI/4.0.3&quot;, &quot;Seurat/3.1.5-R-4.0.0&quot;],
        &quot;tsv_to_bam&quot;:[&quot;subset-bam/1.0&quot;],
        &quot;filter_UMIs&quot;:[&quot;GCC/7.3.0-2.30&quot;,  &quot;OpenMPI/3.1.1&quot;, &quot;Pysam/0.15.1-Python-3.6.6&quot;],
        &quot;bam_to_fastq&quot;:[&quot;bamtofastq/1.2.0&quot;],
        &quot;map_cluster&quot;:[&quot;GCC/5.4.0-2.26&quot;, &quot;OpenMPI/1.10.3&quot;, &quot;STAR/2.6.0c&quot;],
        &quot;TE_count&quot;:[&quot;icc/2018.1.163-GCC-6.4.0-2.28&quot;, &quot;OpenMPI/2.1.2&quot;, &quot;TEToolkit/2.0.3-Python-2.7.14&quot;],
        &quot;TE_count_unique&quot;: [&quot;GCC/7.3.0-2.30&quot;, &quot;OpenMPI/3.1.1&quot;, &quot;Subread/1.6.3&quot;],
        &quot;normalize_TE_counts&quot;:[&quot;GCC/9.3.0&quot;, &quot;OpenMPI/4.0.3&quot;, &quot;Seurat/3.1.5-R-4.0.0&quot;],
        &quot;plot_TE_expression&quot;:[&quot;GCC/9.3.0&quot;, &quot;OpenMPI/4.0.3&quot;, &quot;Seurat/3.1.5-R-4.0.0&quot;]
}
</code></pre><p>If you don&rsquo;t need to load any modules for a software to be available, you can leave the respective list empty.</p>
<hr>
<p>You can read the functions&rsquo; documentation and some tutorials at <a href="https://ra7555ga-s.github.io/truster/">https://ra7555ga-s.github.io/truster/</a></p>

  
  
<style>
	ul.share-buttons{
	  list-style: none;
	  padding: 0;
	}

	ul.share-buttons li{
	  display: inline;
	}

	ul.share-buttons .sr-only{
	  position: absolute;
	  clip: rect(1px 1px 1px 1px);
	  clip: rect(1px, 1px, 1px, 1px);
	  padding: 0;
	  border: 0;
	  height: 1px;
	  width: 1px;
	  overflow: hidden;
	}
</style>


  
<div class="prev-next-post pure-g">
  <div class="pure-u-1-24" style="text-align: left;">
    
    <a href="https://raquelgarza.github.io/github_pages_truster_test/post/migrate-from-jekyll/"><i class="fa fa-chevron-left"></i></a>
    
  </div>
  <div class="pure-u-10-24">
    
    <nav class="prev">
      <a href="https://raquelgarza.github.io/github_pages_truster_test/post/migrate-from-jekyll/">Workflow</a>
    </nav>
    
  </div>
  <div class="pure-u-2-24">
    &nbsp;
  </div>
  <div class="pure-u-10-24">
    
  </div>
  <div class="pure-u-1-24" style="text-align: right;">
    
  </div>
</div>


  
  
  
  
<div id="disqus_thread"></div>
<script type="text/javascript">

(function() {
    
    
    if (window.location.hostname == "localhost")
        return;

    var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
    var disqus_shortname = 'Your Disqus shortname';
    dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
    (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>


  

</div>

</div>
</div>
<script src="https://raquelgarza.github.io/github_pages_truster_test/js/ui.js"></script>
<script src="https://raquelgarza.github.io/github_pages_truster_test/js/menus.js"></script>






<script async src="https://www.googletagmanager.com/gtag/js?id=Your%20Google%20Analytics%20tracking%20ID"></script>
<script>
  
  if (window.location.hostname != "localhost") {
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'Your Google Analytics tracking ID');
  }
</script>








</body>
</html>

